feature/gribi/otg_tests/hierarchical_weight_resolution_test/hierarchical_weight_resolution_test.go

```diff
@@ -0,0 +1,739 @@
+	if deviations.ExplicitInterfaceInDefaultVRF(d) {
+		if a.numSubIntf == 0 {
+			fptest.AssignToNetworkInstance(t, d, p.Name(), deviations.DefaultNetworkInstance(d), 0)
+		} else {
+			for i := uint32(1); i <= a.numSubIntf; i++ {
+				fptest.AssignToNetworkInstance(t, d, p.Name(), deviations.DefaultNetworkInstance(d), i)
+			}
+		}
+	}
+}
+
+// configurePBF returns a fully configured network-instance PF struct.
+func configurePBF(dut *ondatra.DUTDevice) *oc.NetworkInstance_PolicyForwarding {
+	d := &oc.Root{}
+	ni := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	pf := ni.GetOrCreatePolicyForwarding()
+	vrfPolicy := pf.GetOrCreatePolicy(policyName)
+	vrfPolicy.SetType(oc.Policy_Type_VRF_SELECTION_POLICY)
+	vrfPolicy.GetOrCreateRule(1).GetOrCreateIpv4().Protocol = oc.UnionUint8(ipipProtocol)
+	vrfPolicy.GetOrCreateRule(1).GetOrCreateAction().NetworkInstance = ygot.String(nonDefaultVRF)
+	return pf
+}
+
+// applyForwardingPolicy applies the forwarding policy on the interface.
+func applyForwardingPolicy(t *testing.T, ingressPort string) {
+	t.Logf("Applying forwarding policy on interface %v ... ", ingressPort)
+	d := &oc.Root{}
+	dut := ondatra.DUT(t, "dut")
+	interfaceID := ingressPort
+	if deviations.InterfaceRefInterfaceIDFormat(dut) {
+		interfaceID = ingressPort + ".0"
+	}
+	pfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).PolicyForwarding().Interface(interfaceID)
+	pfCfg := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut)).GetOrCreatePolicyForwarding().GetOrCreateInterface(interfaceID)
+	pfCfg.ApplyVrfSelectionPolicy = ygot.String(policyName)
+	pfCfg.GetOrCreateInterfaceRef().Interface = ygot.String(ingressPort)
+	pfCfg.GetOrCreateInterfaceRef().Subinterface = ygot.Uint32(0)
+	if deviations.InterfaceRefConfigUnsupported(dut) {
+		pfCfg.InterfaceRef = nil
+	}
+	gnmi.Replace(t, dut, pfPath.Config(), pfCfg)
+}
+
+// configureATE configures Ethernet + IPv4 on the ATE. If the number of
+// Subinterfaces(numSubIntf) > 0, we then create additional sub-interfaces
+// each with a unique VlanID starting from 1. The IPv4 addresses start with
+// ATE:Port.IPv4 and then nextIP(ATE:Port.IPv4, 4) for each sub interface.
+func (a *attributes) configureATE(t *testing.T, top gosnappi.Config, ate *ondatra.ATEDevice) {
+	t.Helper()
+	p := ate.Port(t, a.Name)
+
+	ip := a.ip(0)
+	gateway := a.gateway(0)
+
+	top.Ports().Add().SetName(p.ID())
+	dev := top.Devices().Add().SetName(a.Name)
+	eth := dev.Ethernets().Add().SetName(a.Name + ".Eth").SetMac(a.MAC)
+	eth.Connection().SetPortName(p.ID())
+	ipObj := eth.Ipv4Addresses().Add().SetName(dev.Name() + ".IPv4")
+	ipObj.SetAddress(ip).SetGateway(gateway).SetPrefix(uint32(a.IPv4Len))
+	t.Logf("Adding ATE Ipv4 address: %s with gateway: %s", cidr(ip, 30), gateway)
+
+	for i := uint32(1); i <= a.numSubIntf; i++ {
+		name := fmt.Sprintf(`dst%d`, i)
+		ip = a.ip(uint8(i))
+		gateway = a.gateway(uint8(i))
+		mac, _ := incrementMAC(a.MAC, int(i)+1)
```

Comment:
Should check error (and use a `t.Fatalf` if there is one.)

Reply:
added